packages/test/test-end-to-end-tests/src/test/gc/gcTestSummaryUtils.ts

```diff
@@ -0,0 +1,177 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ContainerRuntimeFactoryWithDefaultDataStore } from "@fluidframework/aqueduct";
+import { assert } from "@fluidframework/common-utils";
+import { IContainer, IHostLoader, LoaderHeader } from "@fluidframework/container-definitions";
+import { ConnectionState } from "@fluidframework/container-loader";
+import {
+    gcBlobPrefix,
+    gcTreeKey,
+    IContainerRuntimeOptions,
+    IGCRuntimeOptions,
+    ISummarizer,
+} from "@fluidframework/container-runtime";
+import { FluidObject, IRequest } from "@fluidframework/core-interfaces";
+import { DriverHeader } from "@fluidframework/driver-definitions";
+import { concatGarbageCollectionStates } from "@fluidframework/garbage-collector";
+import { ISummaryTree, SummaryType } from "@fluidframework/protocol-definitions";
+import {
+    IContainerRuntimeBase,
+    IFluidDataStoreFactory,
+    IGarbageCollectionState,
+} from "@fluidframework/runtime-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestContainerConfig, ITestObjectProvider } from "@fluidframework/test-utils";
+import { mockConfigProvider } from "./gcTestConfigs";
+
+const summarizerClientType = "summarizer";
+
+async function createSummarizerCore(container: IContainer, loader: IHostLoader, summaryVersion?: string) {
+    const absoluteUrl = await container.getAbsoluteUrl("");
+    if (absoluteUrl === undefined) {
+        throw new Error("URL could not be resolved");
+    }
+
+    const request: IRequest = {
+        headers: {
+            [LoaderHeader.cache]: false,
+            [LoaderHeader.clientDetails]: {
+                capabilities: { interactive: false },
+                type: summarizerClientType,
+            },
+            [DriverHeader.summarizingClient]: true,
+            [LoaderHeader.reconnect]: false,
+            [LoaderHeader.version]: summaryVersion,
+        },
+        url: absoluteUrl,
+    };
+    const summarizerContainer = await loader.resolve(request);
+    await waitForContainerConnection(summarizerContainer);
+
+    const fluidObject =
+        await requestFluidObject<FluidObject<ISummarizer>>(summarizerContainer, { url: "_summarizer" });
+    if (fluidObject.ISummarizer === undefined) {
+        throw new Error("Fluid object does not implement ISummarizer");
+    }
+    return fluidObject.ISummarizer;
+}
+
+export async function createSummarizerFromFactory(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    dataStoreFactory: IFluidDataStoreFactory,
+    summaryVersion?: string,
+    containerRuntimeFactory = ContainerRuntimeFactoryWithDefaultDataStore,
```

Comment:
```suggestion
    containerRuntimeFactoryType = ContainerRuntimeFactoryWithDefaultDataStore,
```
I think adding the `Type` naming is clearer since we initiate an object of this type below.

Reply:
changed i.t